// mysqlConection/index.ts
import { createPool, Pool, QueryError, ResultSetHeader, RowDataPacket } from "mysql2/promise";

export interface IQuery {
    sql: string;
    values: any[];
}

// 🔹 Variável global para armazenar a pool de conexões
let pool: any = null;

// 🔹 Função para obter a conexão reutilizável
export const getDatabaseConnection = async (): Promise<(sql: string, values: any[]) => Promise<any>> => {
    if (!pool) {
        console.log("🛠️ Criando nova conexão no Lambda Layer...");
        pool = createPool({
            host: "host.docker.internal",
            port: 3306,
            user: "root",
            password: "root",
            database: "teste",
            waitForConnections: true,
            connectionLimit: 10,
            queueLimit: 0,
        });
    } else {
        console.log("♻️ Reutilizando conexão do Layer...");
    }

    return async (sql: string, values: any[]): Promise<any> => {
        try {
            const [results] = await pool.query(sql, values);
            return results;
        } catch (error) {
            throw error;
        }
    };
};
